ocaml/src/asd_config.ml

```diff
@@ -50,3 +49,19 @@ let read_cfg cfg_file =
         "Interpreted the config as: %s"
         ([%show : Config.t] cfg)) >>= fun () ->
   Lwt.return config
+
+let retrieve_cfg_from_file cfg_file =
+  Lwt_extra2.read_file cfg_file >>= fun txt ->
+  Lwt_log.debug_f "Found the following config: %s" txt >>= fun () ->
+  retrieve_cfg_from_string txt
+
+
+
+let retrieve_cfg cfg_url =
+  let open Prelude.Url in
+  Lwt_io.eprintlf "config %s" (show cfg_url) >>= fun () ->
+  match cfg_url with
```

Comment:
These are different configs, but I'll create a function
`retrieve_from_url: url -> (string -> 'a) -> 'a Lwt.t `
to split out the fetching.